requests/adapters.py

```diff
@@ -388,7 +388,10 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                     low_conn.endheaders()
 
                     for i in request.body:
-                        low_conn.send(hex(len(i))[2:].encode('utf-8'))
+                        chunk_size = len(i)
+                        if chunk_size == 0:
+                            break
```

Comment:
So if someone gives us a generator like:

```py
(chunk for chunk in b'foo', b'bar', b'', b'biz')
```

I think they'll expect it all to be sent. So I agree with @Lukasa. That said, we definitely need to also document this to death. I am willing to bet someone at sometime will want that to be a break for some bizarre reason. =P